### ISSUE TITLE
How to load TRT engine model without loading torch model weights?

### ISSUE THREAD

**Issue by swagh1611** [NVIDIA/Isaac-GR00T#340]
I’m trying to deploy a GR00T policy using a TensorRT engine. In my case, I don’t want to load the pretrained PyTorch model weights at all.

I just want to:
1. Instantiate the model architecture (backbone + action head)
2. Load metadata for normalisation
3. Use TensorRT for inference

Is there an officially supported way (or recommended pattern) to load a GR00T policy with just the architecture + metadata, without requiring PyTorch model weights, when the actual inference will be handled by TensorRT?

Thanks!

**Comment by swagh1611** [comment]
I managed to get it working.

In addition to the other TensorRT engines, I had to save both position_embedding.engine and future_tokens.engine. Then, I wrote a TensorRT policy wrapper. With this setup, I now have a standalone TensorRT engine policy running without needing to load Torch weights. I also compared the TensorRT outputs with the pytroch weights, and the results are very similar.

If anyone wants to know how to do this, I can add some code in the future. For now, I’ll close the issue.

**Comment by kc645** [comment]
Hi! I am very interested in this. could you provide guidance / code for this?

**Comment by swagh1611** [comment]
Hi,

Please take a look at the changes I’ve made in the fork [here](https://github.com/Extend-Robotics/Isaac-GR00T/tree/1-add-ER-modality-class/deployment_scripts)

My goals were to:

* **Support a variable number of cameras**.

* **Enable a standalone TensorRT policy engine**.

### Handling variable number of cameras

I followed the guidance from [this issue](https://github.com/NVIDIA/Isaac-GR00T/issues/237)
This required modifications to both `export_onnx.py` and `build_engine.sh`.

### Standalone TensorRT engine

The original code relies on `position_embedding` and `future_tokens` from torch weights.
To enable standalone inference, I added support for exporting these modules to ONNX within `export_onnx.py` and ensured that the corresponding TensorRT engine is generated in `build_engine.sh`.

**Additional changes**

In [export_onnx.py](https://github.com/Extend-Robotics/Isaac-GR00T/blob/1-add-ER-modality-class/deployment_scripts/export_onnx.py):
* Copy `experiment_cfg` and `config.json` from checkpoints folder to **gr00t_onnx** path

In [build_engine.sh](https://github.com/Extend-Robotics/Isaac-GR00T/blob/1-add-ER-modality-class/deployment_scripts/build_engine.sh):
* Added steps to copy the `experiment_cfg` directory and `config.json` from **gr00t_onnx** to **gr00t_engine**, which contain the dataset stats and model hyperparameters respectively required for inference.

**Policy wrapper**
I created a GR00TEnginePolicy (see the code [here](https://github.com/Extend-Robotics/Isaac-GR00T/blob/1-add-ER-modality-class/deployment_scripts/trt_policy.py)) to handle inference.
For readability, I also added [config_utils](https://github.com/Extend-Robotics/Isaac-GR00T/blob/1-add-ER-modality-class/deployment_scripts/config_utils.py) to load `config.json`.

**Testing**
You can test the pipeline using [gr00t_inference_trt.py](https://github.com/Extend-Robotics/Isaac-GR00T/blob/1-add-ER-modality-class/deployment_scripts/gr00t_inference_trt.py).


Let me know if you need further help or something is not clear.

**Comment by kc645** [comment]
It worked!! thank you!!

**Comment by FanZhang91** [comment]
The link has expired. where is the code? @swagh1611 @kc645
